### ISSUE TITLE
Lint for replacing `x.strip_[prefix/suffix](...).unwrap_or(x)` with `x.trim_[prefix/suffix](...)`

### ISSUE THREAD

**Issue by ChayimFriedman2** [rust-lang/rust-clippy#17779]
### What it does

Rust 1.100.0 will stabilize the [`str::trim_prefix()`](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.trim_prefix) and `trim_suffix()` methods. The previous workaround for them was `s.strip_[prefix/suffix](...).unwrap_or(s)`. I think we should have a lint for the new method.

### Advantage

It's just simpler, and perhaps also faster.

### Example

```rust
let s: &str;
s.strip_prefix("foo").unwrap_or(s)
```

Could be written as:

```rust
let s: &str;
s.trim_prefix("foo")
```

**Comment by CommanderStorm** [comment]
@rustbot label good first issue

That sounds like a good and simple `methods` lint to me. Needs MSRV-checking though, but that is not hard ;)
